src/pages/TvShowsPage/TvShowsPage.tsx

```diff
@@ -0,0 +1,21 @@
+import './TvShowsPage.css'
+import { AppLayout } from "../../components/layout/AppLayout/AppLayout.tsx";
+import { useLocation } from "react-router-dom";
+import { usePathNameState } from "../../states/pathNameState.ts";
+import { useEffect } from "react";
+
+export const TvShowsPage = () => {
+    const location = useLocation()
+    const { setPathName } = usePathNameState()
+
+    useEffect(() => {
+        setPathName(location.pathname)
+    }, [location.pathname, setPathName])
```

Comment:
these lines are repeated in several components. It's better to put them in separate custom hook